### PR TITLE
Avoid config change to be lost.

### PR DESCRIPTION
**Pull request by shigui1989**

### CODE REVIEW

File: src/handle_join_leave.cxx [hunk 1]
```diff
@@ -210,6 +210,17 @@ void raft_server::sync_log_to_new_srv(ulong start_idx) {
               gap, quick_commit_index_.load(), start_idx,
               params->log_sync_stop_gap_ );
 
+        // See comment for raft_params::interval_to_wait_previous_config_change_finish_.
+        int32 count = 0;
+        while (config_changing_) {
+            p_in( "Try to add server %d. But now there is a "
+                  "un-committed config change. Should wait. "
+                  "Count: %d.", srv_to_join_->get_id(), count++);
+            std::this_thread::sleep_for
```
[the remark this answers is not in the document]
Comment (shigui1989) [reply src/handle_join_leave.cxx]: Yes. You can help. Please go ahead! Thanks.

[the remark this answers is not in the document]
Comment (shigui1989) [reply src/handle_join_leave.cxx]: @greensky00

[the remark this answers is not in the document]
Comment (greensky00) [reply src/handle_join_leave.cxx]: Done.

[the remark this answers is not in the document]
Comment (shigui1989) [reply src/handle_join_leave.cxx]: Thanks

**Pull request merged**